py/smallint.h

```diff
@@ -61,6 +61,13 @@
 
 #define MP_SMALL_INT_MAX ((mp_int_t)(~(MP_SMALL_INT_MIN)))
 
+// https://stackoverflow.com/a/4589384/1976323
+/* Number of bits in inttype_MAX, or in any (1<<k)-1 where 0 <= k < 2040 */
+#define _IMAX_BITS(m) ((m) / ((m) % 255 + 1) / 255 % 255 * 8 + 7 - 86 / ((m) % 255 + 12))
```

Comment:
I don't think this should start with `_` because such names are reserved for the compiler.  Maybe `MP_IMAX_BITS`?

Reply:
done